scripts/check_changelog_fragments.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+# This script is intended to run during CI, however it can be run locally by
+# committing changelog fragments before executing the script. If the script
+# finds an issue with your changelog fragment, you can un-stage the fragment
+# from being committed and fix the issue.
+
+CHANGELOG_DIR="changelog.d"
+FRAGMENT_TYPES="breaking|security|deprecation|feature|enhancement|fix"
+
+if [ ! -d "${CHANGELOG_DIR}" ]; then
+  echo "No ./${CHANGELOG_DIR} found. This tool must be invoked from the root of the repo."
+  exit 1
+fi
+
+# diff-filter=A lists only added files
+FRAGMENTS=$(git diff --name-only --diff-filter=A origin/master ${CHANGELOG_DIR})
+
+if [ "$(echo "$FRAGMENTS" | grep -c .)" -lt 1 ]; then
+  echo "No changelog fragments detected"
+  echo "If no changes  necessitate user-facing explanations, add the GH label 'no-changelog'"
+  echo "Otherwise, add changelog fragments to changelog.d/"
```

Comment:
I might point people to the added `./changelog.d/README.md` file. I think this would help contributors more easily discover it.